metaflow/plugins/kfp/kfp.py

```diff
@@ -105,6 +106,61 @@ def _get_retries(node):
 
         return max_user_code_retries, max_user_code_retries + max_error_retries
 
+    @staticmethod
+    def _get_resource_requirements(node):
+        """
+        Get resource request or limit for a Metaflow step (node).
+        If @resources is also present, the maximum value is used for resource request, and lowest
+        value is used for resource limit.
+
+        Eventually resource request and limits link back to kubernetes, see
+        https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/
+
+        Supported parameters: 'cpu', 'cpu_limit', 'gpu', 'gpu_vendor', 'memory', 'memory_limit'
+        Keys with no suffix set resource request (minimum);
+        keys with 'limit' suffix sets resource limit (maximum).
+
+        Default unit for memory is megabyte.
+
+        Example using resource decorator:
+            @resource(cpu=0.5, cpu_limit=2, gpu=1, memory=300)
+            @step
+            def my_kfp_step(): ...
+        """
+
+        resource = dict()
+        for deco in node.decorators:
+            if isinstance(deco, ResourcesDecorator):
+                for k, v in deco.attributes.items():
+                    my_val = resource.get(k)
+
+                    if v is not None:
+                        if my_val is not None:
+                            # Resource decorator should only be used once to avoid confusion
+                            # This only happens when user statically define same resource
+                            # requirement with more than one resource decorator for a single step.
+                            # Decorators attached during runtime doesn't fall into this category
+                            # for being dropped by metaflow.decorators._attach_decorators_to_step.
+                            # TODO before merging: should this print or raise error?
+                            print(
+                                f"Resource requirement {k} is set more than once. "
+                                f"The first request with value {my_val} is taken "
+                                f"over the second request with value {v}."
+                            )
+
+                        else:
+                            v = str(v)
+
+                            if k == "gpu" and v == "0":  # Ignore 0 gpu request
+                                continue
+
+                            if k in ["memory", "memory_limit"] and v.isnumeric():
+                                v += "M"  # Metaflow default unit for memory is Megabyte
+
+                            resource[k] = v
```

Comment:
I'd make this section of code a translator of @resource dictionary to a dictionary that is of K8s language (example: adding "M").